libsolidity/formal/Predicate.cpp

```diff
@@ -381,7 +384,7 @@ std::vector<std::optional<std::string>> Predicate::summaryPostInputValues(std::v
 
 	auto const& inParams = function->parameters();
 
-	auto first = _args.begin() + 6 + static_cast<int>(stateVars->size()) * 2 + static_cast<int>(inParams.size()) + 1;
+	auto first = _args.begin() + (long)firstArgIndex() + static_cast<int>(stateVars->size()) * 2 + static_cast<int>(inParams.size()) + 1;
```

Comment:
because return type of firstArgIndex() is size_t (which seemed to me it makes most sense)
and operator `+` for iterator takes long as an argument

Reply:
Since the other offsets use `static_cast<int>` maybe we should at least be consistent with that?
In general, we should prefer `static_cast` from C++ to C-style casts.

Reply:
Right! It's being used in the same line and I didn't notice it!